packages/components/pie-toast-provider/test/component/pie-toast-provider.spec.ts

```diff
@@ -146,4 +146,58 @@ test.describe('PieToastProvider - Component tests', () => {
             });
         });
     });
+
+    test.describe('Interactivity', () => {
+        test('should keep page interactive when toast is displayed', async ({ page }) => {
+            // Arrange
+            const pieToastProviderPage = new BasePage(page, 'toast-provider--scroll-page');
+            await pieToastProviderPage.load();
+
+            const expectedEventMessage = 'Section button clicked';
+
+            const consoleMessages: string[] = [];
+            page.on('console', (message) => {
+                if (message.type() === 'info') {
+                    consoleMessages.push(message.text());
+                }
+            });
+
+            const toastElement = page.locator('pie-toast');
+            await expect(toastElement).toBeVisible();
+
+            // Act
+            const sectionButton = page.locator('pie-button').filter({ hasText: 'Interactive element' });
+            await expect(sectionButton).toBeVisible();
+            await sectionButton.click();
+
+            // Assert
+            expect(consoleMessages).toEqual([expectedEventMessage]);
+        });
+    });
+
+    test.describe('Position and Scrolling', () => {
+        test('should maintain fixed position when scrolling', async ({ page }) => {
+            // Arrange
+            const pieToastProviderPage = new BasePage(page, 'toast-provider--scroll-page');
+            await pieToastProviderPage.load();
+
+            const toastElement = page.locator('pie-toast');
+            await expect(toastElement).toBeVisible();
+
+            const initialPosition = await toastElement.boundingBox();
+
+            // Act
+            await page.evaluate(() => {
+                window.scrollTo(0, document.body.scrollHeight);
+            });
+
+            // Assert
+            const finalPosition = await toastElement.boundingBox();
+
+            if (initialPosition && finalPosition) {
```

Comment:
Is it worth adding an else condition here that executes `test.fail("message");` in the event these don't have values?

Reply:
The check is only there to satisfy TypeScript, as the resolved promise can be `null`. I've tried to use `test.fail` but it it expects a test body again which feels redundant. I have updated the test to use optional chaining instead and that should make the test fail if the values ever become null